presto-client/src/main/java/com/facebook/presto/client/StatementClient.java

```diff
@@ -351,7 +351,18 @@ public void close()
                         .setHeader(USER_AGENT, USER_AGENT_VALUE)
                         .setUri(uri)
                         .build();
-                httpClient.executeAsync(request, createStatusResponseHandler());
+                // at this point the current thread may be interrupted and if it is, below http call may fail silently
+                // so clear the interrupt flag and recover the flag when the http call is completed
+                Thread currentThread = Thread.currentThread();
+                boolean isInterrupted = currentThread.interrupted();
+                try {
+                    httpClient.execute(request, createStatusResponseHandler());
+                }
```

Comment:
I converted this to a synchronous call to get rid of the possibility of the cli main thread not seeing its interrupted status even though it's interrupted (although this is probably a small window of vulnerability). Because, if it was an asynchronous call I would recover the interrupt flag when the future is done, but before the future is done `close()` method can return and the main thread may check its interrupted status and not see that it's interrupted.